File: java/Transaksi.java
public class Transaksi {
    // =========== Atribut Private =========== //
    private int idTransaksi;
    private int idPembeli;
    private int idProduk;
    private String tanggal;
    private double totalHarga;
    private int jumlahItem;

    // =========== Constructor =========== //
    public Transaksi(int idTransaksi, int idPembeli, int idProduk, String tanggal, int jumlahItem, double totalHarga) {
        this.idTransaksi = idTransaksi;
        this.idPembeli = idPembeli;
        this.idProduk = idProduk;
        this.tanggal = tanggal;
        this.jumlahItem = jumlahItem;
        this.totalHarga = totalHarga;
    }

    // =========== Setter =========== //
    public void setIdTransaksi(int idTransaksi) {
        this.idTransaksi = idTransaksi;
    }

    public void setIdPembeli(int idPembeli) {
        this.idPembeli = idPembeli;
    }

    public void setIdProduk(int idProduk) {
        this.idProduk = idProduk;
    }

    public void setTanggal(String tanggal) {
        this.tanggal = tanggal;
    }

    public void setJumlahItem(int jumlahItem) {
        this.jumlahItem = jumlahItem;
    }

    public void setTotalHarga(double totalHarga) {
        this.totalHarga = totalHarga;
    }

    // =========== Getter =========== //
    public int getIdTransaksi() {
        return this.idTransaksi;
    }

    public int getIdPembeli() {
        return this.idPembeli;
    }

    public int getIdProduk() {
        return this.idProduk;
    }

    public String getTanggal() {
        return this.tanggal;
    }

    public int getJumlahItem() {
        return this.jumlahItem;
    }

    public double getTotalHarga() {
        return this.totalHarga;
    }

    // =========== Method untuk menampilkan info =========== //
    public void displayInfo() {
        System.out.println("+----------------------------------+");
        System.out.println("|         INFORMASI TRANSAKSI      |");
        System.out.println("+----------------------------------+");
        System.out.println("| ID Transaksi : " + idTransaksi);
        System.out.println("| ID Pembeli   : " + idPembeli);
        System.out.println("| ID Produk    : " + idProduk);
        System.out.println("| Tanggal      : " + tanggal);
        System.out.println("| Jumlah Item  : " + jumlahItem);
        System.out.println("| Total Harga  : Rp " + totalHarga);
        System.out.println("+----------------------------------+\n");
    }

}

File: cpp/Transaksi.cpp
#include <iostream>
using namespace std;

class Transaksi {
    private:
        int idTransaksi;
        int idPembeli;
        int idProduk;
        string tanggal;
        double totalHarga;
        int jumlahItem;
    public:
        // =========== Constructor =========== //
        Transaksi(int idTransaksi, int idPembeli, int idProduk, string tanggal, int jumlahItem, int totalHarga) {
            this->idTransaksi = idTransaksi;
            this->idPembeli = idPembeli;
            this->idProduk = idProduk;
            this->tanggal = tanggal;
            this->jumlahItem = jumlahItem;
            this->totalHarga = totalHarga;
        }

        //============ Setter ============//
        // method untuk mengatur id transaksi
        void setIdTransaksi(int idTransaksi) {
            this->idTransaksi = idTransaksi;
        }   

        // method untuk mengatur id pembeli
        void setIdPembeli(int idPembeli) {
            this->idPembeli = idPembeli;
        }   


        // method untuk mengatur id produk
        void setIdProduk(int idProduk) {
            this->idProduk = idProduk;
        }

        // method untuk mengatur tanggal transaksi
        void setTanggal(string tanggal) {
            this->tanggal = tanggal;
        } 

        // method untuk mengatur jumlah item
        void setJumlahItem(int jumlahItem) {
            this->jumlahItem = jumlahItem;
        }   

        // method untuk mengatur total harga
        void setTotalHarga(double totalHarga) {
            this->totalHarga = totalHarga;
        }

        //============ Getter ============//
        // method untuk mendapatkan id transaksi    
        int getIdTransaksi() {
            return this->idTransaksi;
        }

        // method untuk mendapatkan id pembeli
        int getIdPembeli() {
            return this->idPembeli;
        }

        // method untuk mendapatkan id produk
        int getIdProduk() { 
            return this->idProduk;
        }

        // method untuk mendapatkan tanggal transaksi
        string getTanggal() {
            return this->tanggal;
        }
          
        // method untuk mendapatkan jumlah item
        int getJumlahItem() {
            return this->jumlahItem;
        }

        // method untuk mendapatkan total harga
        double getTotalHarga() {
            return this->totalHarga;
        }

        // method untuk menampilkan informasi transaksi
        void displayInfo() {
            cout << "+----------------------------------+\n";
            cout << "|         INFORMASI TRANSAKSI      |\n";
            cout << "+----------------------------------+\n";
            cout << "| ID Transaksi : " << idTransaksi << "\n";
            cout << "| ID Pembeli   : " << idPembeli << "\n";
            cout << "| ID Produk    : " << idProduk << "\n";
            cout << "| Tanggal      : " << tanggal << "\n";
            cout << "| Jumlah Item  : " << jumlahItem << "\n";
            cout << "| Total Harga  : Rp " << totalHarga << "\n";
            cout << "+----------------------------------+\n\n";
        }


        // ========== Destructor =========== //
        ~Transaksi() {
            // Destructor
        }
};